tests/testthat/test_oldnew.R

```diff
@@ -89,6 +89,18 @@ test_that("matches old data system output", {
       olddata <- COMPDATA[[oldf]]
       expect_is(olddata, "data.frame", info = paste("No comparison data found for", oldf))
 
+      # During the base year update development process, some outputs from the data
+      # system will be extended and therefore have different dimensions from the
+      # old comparison data. This causes the old-new test to fail.
+      # The extended extrapolation years will be dropped from the newdata
+      # so that comparison can be made to the old data. This also serves to check
+      # that the extrapolation procedure does not touch original data.
+      if(max(newdata$year) == BYU_YEAR){ # one way to check that it's a BYU without flags.
+        newdata %>%
+          filter(year <= max(HISTORICAL_YEARS)) ->
```

Comment:
I think the `if` will do the trick but this `filter` is not the right way to handle the correction to get the dimensions to agree. So travis is still failing. 

The problem is that the last year of data in an input file may not actually be the `max(HISTORICAL_YEARS)`. For the GDP file, the last recorded year is 2013, not 2010. From the travis failures, it looks like there's probably 4 more files that happen to have the last recorded year = 2015 (`BYU_YEAR`) and not 2010 and so are falling into this `if` statement as well. There could be even more that have a recorded year like 2012 or 2013 but aren't yet BYU'd to 2015. 

I cannot think of an endogenous way to detect what the last recorded year was before extrapolation without adding additional information. I think I have an idea to incorporate it into a new FLAG with some `if` statements _in each zchunk that is BYU'd_ to handle it. Which should be fine but is also an extra place for the teams to make a mistake, even if we present careful notes and examples. Obviously I'd put some thought into minimizing that. 

Alternatively, if we don't think this part of the test will be kept beyond the actual base year updating process, we could also do 
```
olddata %>%
          filter(year <= max(HISTORICAL_YEARS)) ->
          olddata 
```
We will lose a few years of data from the oldnew test but there's less space for user error. 


So, do we want to go flag or slightly less robust oldnew test?

Reply:
There were only 10 or so that failed.  Could we just flag them as `FLAG_NO_TEST` for now?

Reply:
One thing to keep in mind, the FAO Aquastat data are typically written out in (at best) 5-year intervals, with the data falling in years like 2008, 2012, 2017. Often there will be 20-year lapses in a time series. So, at least for that specific data source we don't want to be filtering years prior to filling in the missing data.

Reply:
@pkyle Thanks! But here we're talking about the test code, not filtering before filling data.

@abigailsnyder @pralitp I'm also leery of adding extra logic/steps the user is responsible for. Another possible way to address this: weaken the oldnew test so it filters `newdata` to only the years present in `olddata` before doing the comparison. This would trim off any extrapolated data; the cost is that now it won't detect bad newdata that have extra years for non-BYU reasons.